app.rb

```diff
@@ -0,0 +1,62 @@
+require 'sinatra/base'
+require 'sinatra/reloader'
+require 'sinatra/flash'
+require_relative './lib/users'
+require_relative './lib/database_connection'
+# require_relative './connect_to_database'
+require './database_connection_setup'
+
+class MakersBnB < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  register Sinatra::Flash
+
+  get '/' do
+    erb(:index)
+  end
+
+  get '/users/new' do
+    erb(:'users/new')
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email], password: params[:password])
+    p @user
+    session[:user_id] = @user.id
+    redirect '/spaces'
+  end
+
+  get '/spaces' do
+    @user = User.find(session[:user_id])
+    erb(:spaces) #Welcome... take the user information fro the user in the session.
+  end
+
+  post '/sessions/destroy' do
```

Comment:
Minor: I feel like maybe this should be at the end, after '/sessions/new' and post '/sessions'. Just for readability